pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -45,14 +44,25 @@ def solve_NLP_subproblem(solve_data, config):
 
     MindtPy.MindtPy_linear_cuts.deactivate()
     fix_nlp.tmp_duals = ComponentMap()
+    # tmp_duals are the value of the dual variables stored before using deactivate trivial contraints
+    # The values of the duals are computed as follows: (Complementary Slackness)
+    #
+    # | constraint | c_leq | status at x1 | tmp_dual  |
+    # |------------|-------|--------------|-----------|
+    # | g(x) <= b  | -1    | g(x1) <= b   | 0         |
+    # | g(x) <= b  | -1    | g(x1) > b    | b - g(x1) |
+    # | g(x) >= b  | +1    | g(x1) >= b   | 0         |
+    # | g(x) >= b  | +1    | g(x1) < b    | b - g(x1) |
+
```

Comment:
The condition for the duals allows us to identify which side of the inequality is active for us to add the cuts.
We prefer to include the upper bound as the right hand side since we are considering c by default a (hopefully) convex function, which would make c >= lb a nonconvex inequality which we wouldn't like to add linearizations if we don't have to.